src/gui/tray/activitylistmodel.cpp

```diff
@@ -542,11 +514,145 @@ void ActivityListModel::activitiesReceived(const QJsonDocument &json, int status
     emit activityJobStatusCode(statusCode);
 }
 
+std::pair<int, int> ActivityListModel::rowRangeForEntryType(const ActivityEntryType type)
+{
+    // We want to present activities in a certain order, and we want to ensure entry types are grouped together.
+    // We therefore need to find the range of rows in the finalList that represent an entry type group.
+    int startRow = 0;
+
+    // Start from the type that we want to push down the furthest. Cascade through switch cases.
+    switch(type) {
+    case ActivityEntryType::MoreActivitiesAvailableType:
+        startRow = _finalList.count();
+        break;
+    case ActivityEntryType::ActivityType:
+        startRow += _syncFileItemLists.count();
+    case ActivityEntryType::SyncFileItemType:
+        startRow += _notificationLists.count();
+    case ActivityEntryType::NotificationType:
+        // We only show one activity for ignored files
+        if(_listOfIgnoredFiles.count() > 0) {
+            startRow += 1;
+        }
+    case ActivityEntryType::IgnoredFileType:
+        startRow += _notificationErrorsLists.count();
+    // Remaining types should go at top
+    default:
+        break;
+    }
+
+    int entryRowCount = -1;
+
+    switch(type) {
+    case ActivityEntryType::ActivityType:
+        entryRowCount = _activityLists.count();
+        break;
+    case ActivityEntryType::SyncFileItemType:
+        entryRowCount = _syncFileItemLists.count();
+        break;
+    case ActivityEntryType::NotificationType:
+        entryRowCount = _notificationLists.count();
+        break;
+    case ActivityEntryType::ErrorType:
+        entryRowCount = _notificationErrorsLists.count();
+        break;
+
+    // Single activity sections
+    case ActivityEntryType::IgnoredFileType:
+        if(_listOfIgnoredFiles.count() <= 0) {
+            break;
+        }
+    case ActivityEntryType::MoreActivitiesAvailableType:
+        if(!_showMoreActivitiesAvailableEntry) {
+            break;
+        }
+    case ActivityEntryType::DummyFetchingActivityType:
+        if(_finalList.count() > 0 && _finalList.first() != _dummyFetchingActivities) {
+            break;
+        }
+
+    // All cascade down to here
+        entryRowCount = 1;
+    default:
+        break;
```

Comment:
please avoid using `default` when enum type is coming from our code because it will make future evolution much harder (no way for the compiler to notify that you missed the new enum values

Reply:
Okay, removed